In [1]:
from sklearn.datasets import load_iris




iris = load_iris()
numSamples, numFeatures = iris.data.shape
print(numSamples)
print(numFeatures)
print(list(iris.target_names))

150
4
['setosa', 'versicolor', 'virginica']


I am going to use the Iris dataset to test out XGBoost.

It is an inbuilt dataset in Jupyter Notebook.

This data set includes the width and length of the petals and sepals of many flowers, and the specific species of Iris the flower belongs to.

There are a total of 150 samples of flowers.

Each flower has 4 features. The length and width of the petals and sepals of the flowers.

Setosa, Versicolor and Virginica are the 3 subspecies of flowers.

In [2]:
from sklearn.model_selection import train_test_split




X_train, X_test, y_train, y_test = train_test_split(iris.data, iris.target, test_size = 0.2, random_state = 0)

Here, the data is getting split into the training and testing. 

20% of the data is going to be used for testing and 80% of the data is going to be used for training.

In [3]:
import xgboost as xgb




train = xgb.DMatrix(X_train, label = y_train)
test = xgb.DMatrix(X_test, label = y_test)

In [8]:
param = {
    'max_depth': 4,
    'eta': 0.3,
    'objective': 'multi:softmax',
    'num_class': 3
}
epochs = 10

The maximum depth of the trees is 4.

The learning rate is 0.3.

Softmax is used here since this is a multiple classification problem.

The number of classifications is tree because there are only 3 subspecies to choose from.

This algorithm is going to iterate 10 times.

In [5]:
model = xgb.train(param, train, epochs)

[08:02:08] WARNING: C:/Users/Administrator/workspace/xgboost-win64_release_1.4.0/src/learner.cc:1095: Starting in XGBoost 1.3.0, the default evaluation metric used with the objective 'multi:softmax' was changed from 'merror' to 'mlogloss'. Explicitly set eval_metric if you'd like to restore the old behavior.


The model has been trained! Now we are going to compare the predictions and the test data.

In [6]:
predictions = model.predict(test)

In [7]:
print(predictions)

[2. 1. 0. 2. 0. 2. 0. 1. 1. 1. 2. 1. 1. 1. 1. 0. 1. 1. 0. 0. 2. 1. 0. 0.
 2. 0. 0. 1. 1. 0.]


These are the predictions from the test dataset.

0: Setosa
1: Versicolor
2: Virginica

Now we must verify if the predictions are correct. To do this we must find the accuracy of the predictions.

In [10]:
from sklearn.metrics import accuracy_score




accuracy_score(y_test, predictions)

1.0

This model is 100% accurate!